### PR TITLE
Set color of text based on contrast

### LINKED ISSUE #658: Bug: White label is not visible
### Describe the bug

White label is not visible

### To Reproduce

1. Create an Issue
1. Create a label with white color
1. Apply the label to the issue

Eg: https://github.com/yg/open-sauced-goals/issues/14

### Expected behavior

Should have the font color automatically based on the label color!

### Screenshots

Screenshot

https://www.loom.com/share/8dfa9235b074446e983efade26370817

### PR DESCRIPTION
**Pull request by filiptronicek**
Co-Authored-By: @bdougie, @yg 

## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [ ] ✨ Feature
- [ ] 👷 Optimization
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description
Change the current font color based on contrast of the background (label) color and the text.

## Related Tickets & Documents
Closes #658 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
labels

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [x] 🙅 no documentation needed

### CODE REVIEW

File: src/components/IssueListItem.js [hunk 1]
```diff
@@ -7,6 +7,10 @@ import {IssueOpenedIcon, CommentIcon, MilestoneIcon} from "@primer/octicons-reac
 
 dayjs.extend(relativeTime);
 
+function fontContrast(color) {
+  return Number("0x" + color) > 16000000 ? "black" : "white";
+}
```
Comment (yg) [inline src/components/IssueListItem.js:12]: @filiptronicek Moving this to `lib` folder would be nice! This helps us to reuse this function everywhere!

File: src/components/IssueListItem.js [hunk 2]
```diff
@@ -7,6 +7,10 @@ import {IssueOpenedIcon, CommentIcon, MilestoneIcon} from "@primer/octicons-reac
 
 dayjs.extend(relativeTime);
 
+function fontContrast(color) {
+  return Number("0x" + color) > 16000000 ? "black" : "white";
```
Comment (yg) [inline src/components/IssueListItem.js:11]: @filiptronicek We need to figure out the correct threshold value `16000000` will not work for all light colors! We should try the values in trial and error method.

Comment (filiptronicek) [reply src/components/IssueListItem.js:11]: I will try to look at any online resources, to get a better picture of what the treshold value should be

Comment (filiptronicek) [reply src/components/IssueListItem.js:11]: Nonetheless, it looks it is doing a pretty good job at least in our repo
image of labels

Comment (yg) [reply src/components/IssueListItem.js]: @filiptronicek Can you check with `chore` label?

Comment (filiptronicek) [reply src/components/IssueListItem.js]: Sure thing!

Comment (filiptronicek) [reply src/components/IssueListItem.js]: It looks like this one is off, it displays it in white, but should in black

Comment (filiptronicek) [reply src/components/IssueListItem.js]: The `threshold` needs to be 13000000 or less for it to be dark

**Comment by filiptronicek** [comment]
This is a tough one, for the text to be white on the red `bug` label, the threshold needs to be >=16M, but for it to be black on `chore`, it needs to be <=13M

**Comment by filiptronicek** [comment]
I think the [scottcorgan/contrast](https://github.com/scottcorgan/contrast) package really helped it. Works with most labels now.

**Comment by filiptronicek** [comment]
https://runkit.com/filiptronicek/5eff1142988aae001ff65dfd to see in action

**Comment by yg** [comment]
Thanks, @filiptronicek It works perfectly :+1:!

**Pull request merged**